azurelinuxagent/ga/update.py

```diff
@@ -311,39 +311,47 @@ def run(self, debug=False):
                 #
                 # Process the goal state
                 #
-                protocol.update_goal_state()
-
-                if self._upgrade_available(protocol):
-                    available_agent = self.get_latest_agent()
-                    if available_agent is None:
-                        logger.info(
-                            "Agent {0} is reverting to the installed agent -- exiting",
-                            CURRENT_AGENT)
-                    else:
-                        logger.info(
-                            u"Agent {0} discovered update {1} -- exiting",
-                            CURRENT_AGENT,
-                            available_agent.name)
-                    break
+                goal_state_fetched = False
+                try:
+                    protocol.update_goal_state()
+                    goal_state_fetched = True
+                except Exception as e:
+                    msg = u"Exception retrieving the goal state: {0}".format(ustr(traceback.format_exc()))
+                    logger.warn(msg)
+                    add_event(AGENT_NAME, op=WALAEventOperation.FetchGoalState, version=CURRENT_VERSION, is_success=False, message=msg)
```

Comment:
Wouldn't `add_event` also log the message if `is_success=False`?

Reply:
thanks; removed